### PR TITLE
COM-289: Refactor theming of `Rte`

### PR DESCRIPTION
**Pull request by andrearutrecht**
<details><summary>Video of Rte:</summary>
<p>

https://github.com/vivid-planet/comet/assets/122883866/6496ca9e-8697-4caf-afae-53d9c7fc4a8c

</p>
</details> 

<details><summary>Video of disabled Rte:</summary>
<p>

https://github.com/vivid-planet/comet/assets/122883866/28563d3f-5ae4-4e5a-82a2-f5b0ace65ea8

</p>
</details>

### CODE REVIEW

File: packages/admin/admin-rte/src/core/Rte.tsx [hunk 1]
```diff
@@ -148,201 +153,232 @@ export const styleMap = {
...
-};
+        ${ownerState.disabled &&
+        css`
+            color: ${theme.palette.text.disabled};
+        `}
+
+        ${ownerState.minHeight &&
+        css`
+            min-height: ${ownerState.minHeight};
+        `}
```
Comment (jamesricky) [inline packages/admin/admin-rte/src/core/Rte.tsx:377]: Shouldn't this be part of `.public-DraftEditor-content`?

File: packages/admin/admin-rte/src/core/Rte.tsx [hunk 2]
```diff
@@ -148,201 +153,232 @@ export const styleMap = {
...
+            ...restProps
+        } = useThemeProps({
+            props: inProps,
+            name: "CometAdminRte",
+        });
 
-    // merge default options with passed options
-    let options = passedOptions ? { ...defaultOptions, ...passedOptions } : defaultOptions;
+        const ownerState: OwnerState = {
+            disabled,
```
Comment (jamesricky) [inline packages/admin/admin-rte/src/core/Rte.tsx:173]: This is missing `minHeight`

File: packages/admin/admin-rte/src/core/Rte.tsx [hunk 3]
```diff
@@ -148,201 +153,232 @@ export const styleMap = {
...
+const Editor = styled("div", {
+    name: "CometAdminRte",
+    slot: "editor",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [styles.editor, ownerState.disabled && styles.disabled];
+    },
+})<{ ownerState: OwnerState }>(
+    ({ ownerState, theme }) => css`
+        & .public-DraftEditor-content {
+            min-height: 240px;
```
Comment (jamesricky) [inline packages/admin/admin-rte/src/core/Rte.tsx:364]: This is different, than the previous implementation: `${props.minHeight === undefined ? 240 : props.minHeight}px`

File: packages/admin/admin-rte/src/core/Rte.tsx [hunk 4]
```diff
@@ -148,201 +153,232 @@ export const styleMap = {
...
+
+        ${ownerState.minHeight &&
+        css`
+            min-height: ${ownerState.minHeight};
+        `}
+    `,
+);
 
-export default withStyles(styles, { name: "CometAdminRte" })(React.forwardRef(Rte));
+export default Rte;
```
Comment (jamesricky) [inline packages/admin/admin-rte/src/core/Rte.tsx:381]: The default export is no longer required, if `Rte` is already exported

**Review by jamesricky — CHANGES_REQUESTED** [review]

**Review by jamesricky — DISMISSED** [review]

**Comment by jamesricky** [comment]
@andrearutrecht please resolve the conflicts

**Review by jamesricky — DISMISSED** [review]

File: packages/admin/admin-rte/src/core/Rte.tsx [hunk 5]
```diff
@@ -149,201 +154,232 @@ export const styleMap = {
...
-    if (options.customBlockMap) {
-        deprecatedCustomBlockMap = options.customBlockMap;
-        delete options.customBlockMap;
-    }
+        const editorRef = React.useRef<DraftJsEditor>(null);
+        const editorWrapperRef = React.useRef<HTMLDivElement>(null);
 
-    cleanBlockTypeMap(options.blocktypeMap); // mutate object and print warning when configuration is wrong
+        // merge default options with passed options
+        let options = rteOptions ? { ...defaultOptions, ...rteOptions } : defaultOptions;
```
Comment (johnnyomair) [inline packages/admin/admin-rte/src/core/Rte.tsx:182]: Please keep the name `passedOptions`. Both default options as well as passed options are valid RTE options.

**Review by johnnyomair — CHANGES_REQUESTED** [review]

File: packages/admin/admin-rte/src/core/Rte.tsx [hunk 6]
```diff
@@ -356,7 +392,7 @@ declare module "@mui/material/styles" {
 
     interface Components {
         CometAdminRte?: {
-            defaultProps?: ComponentsPropsList["CometAdminRte"];
+            defaultProps?: Partial<ComponentsPropsList["CometAdminRte"]>;
```
Comment (jamesricky) [inline packages/admin/admin-rte/src/core/Rte.tsx:395]: This is already defined as `Partial` in `ComponentsPropsList`

```suggestion
            defaultProps?: ComponentsPropsList["CometAdminRte"];
```

**Review by jamesricky — CHANGES_REQUESTED** [review]

**Review by jamesricky — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**